### ISSUE TITLE
search support for searching filenames

### ISSUE THREAD

**Issue by inbjo** [go-gitea/gitea#26299]
### Feature Description

Hello, everyone.
I'm using elasticsearch as a search driver.  

I often come across scenarios where I need to search for a file name in a repository or an organization.
Currently, gitea only supports searching for code content.

When I look at the storage structure in elasticsearch, I see that gitea treats the file path as the primary key, and it seems that the primary key cannot be searched for in elasticsearch.

To summarize:
I want to be able to search the file name to find out which repository the file belongs to and the path. I think this is a very useful feature. Many thanks to the gitea team!

**Comment by KN4CK3R** [comment]
You are looking for this
![grafik](https://github.com/go-gitea/gitea/assets/1666336/87d90759-646d-4989-939e-6a50260d3f46)

**Comment by inbjo** [comment]
> You are looking for this ![grafik](https://user-images.githubusercontent.com/1666336/258202299-87d90759-646d-4989-939e-6a50260d3f46.png)

This is fine for finding files in a single repository, but not if the files are spread across different repositories.

**Comment by lunny** [comment]
This should be resolved by #32210
